src/frontend/apps/drive/src/features/explorer/api/useMoveItem.tsx

```diff
@@ -16,73 +15,44 @@ export const useMoveItems = () => {
       await driver.moveItems(payload.ids, payload.parentId);
     },
     onMutate: async (payload: MoveItemPayload) => {
-      const itemIds = payload.ids;
-      // Cancel any outgoing refetches
+      // Cancel any outgoing refetches to avoid overwriting optimistic updates
       await queryClient.cancelQueries({
         queryKey: ["items", payload.oldParentId, "children"],
       });
 
       await queryClient.cancelQueries({
         queryKey: ["items", payload.parentId, "children"],
       });
-
-      // Snapshot the previous value
-      const previousItems: Item[] =
-        queryClient.getQueryData(["items", payload.oldParentId, "children"]) ??
-        [];
-
-      const nextItems: Item[] =
-        queryClient.getQueryData(["items", payload.parentId, "children"]) ?? [];
-
-      // Get the nodes from previous items that match the ids
-      const movedItems = previousItems?.filter((item: Item) =>
-        itemIds.includes(item.id)
-      );
-
-      const newOldParentItems = previousItems?.filter(
-        (item: Item) => !itemIds.includes(item.id)
-      );
-
-      const newNextItems = [...movedItems, ...nextItems];
-
-      // Sort newNextItems by type first (folders first), then by creation date in descending order
-      newNextItems.sort((a, b) => {
-        // First sort by type (folders first)
-        if (a.type !== b.type) {
-          return a.type === ItemType.FOLDER ? -1 : 1;
-        }
-
-        // Then sort by creation date in descending order (newest first)
-        const dateA = new Date(a.created_at || 0).getTime();
-        const dateB = new Date(b.created_at || 0).getTime();
-        return dateB - dateA;
+    },
+    onSuccess: (data, payload: MoveItemPayload) => {
+      // Invalidate queries after successful mutation
+      // This will trigger refetch if queries are currently mounted/observed
+      queryClient.invalidateQueries({
+        queryKey: ["items", payload.oldParentId, "children", "infinite"],
       });
 
-      // Optimistically update to the new value
-      queryClient.setQueryData(
-        ["items", payload.oldParentId, "children"],
-        () => newOldParentItems
-      );
+      queryClient.invalidateQueries({
+        queryKey: ["items", payload.parentId, "children", "infinite"],
+      });
 
```

Comment:
You removed the optimistic update. Was it expected? 

It changes the snappyness of the move feature